### PR TITLE
[lit] Automatically detect `customElements.define()` and `@customElement()` calls

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Removes the need for manual `export const tagName = "my-element";`
- Supports `customElements.define("my-element", MyElement)` calls
- Supports Lit's `@customElements("my-element")` decorator

## Testing

Manually, updating the `framework-lit` example

## Docs

Will do be before merge.

### CODE REVIEW

File: examples/framework-lit/src/components/Counter.js
Comment (natemoo-re) [inline examples/framework-lit/src/components/Counter.js:3]: These can be removed, they're no longer needed!

File: examples/framework-lit/src/components/Counter.js
Comment (natemoo-re) [inline examples/framework-lit/src/components/Counter.js:32]: Our custom Lit Vite plugin will pick up on this and add an `export const __astroTagName = "my-counter"` export

File: examples/framework-lit/src/components/Greeting.ts
Comment (natemoo-re) [inline examples/framework-lit/src/components/Greeting.ts:4]: Our custom Lit Vite plugin will also pick up on the `@customElement()` Lit decorator!

File: examples/framework-lit/src/components/Test.js
Comment (natemoo-re) [inline examples/framework-lit/src/components/Test.js:18]: This also works!

File: packages/renderers/renderer-lit/vite-plugin-lit.js
Comment (natemoo-re) [inline packages/renderers/renderer-lit/vite-plugin-lit.js:98]: This is pretty much it. If we find `@customElement("my-element")` or `customElements.define("my-element", MyElement)`, we add the export while respecting sourcemaps.

File: packages/renderers/renderer-lit/vite-plugin-lit.js
Comment (natemoo-re) [inline packages/renderers/renderer-lit/vite-plugin-lit.js:33]: Parse with `acorn`.

**Comment by matthewp** [comment]
This is definitely very cool, but I think we should talk about this with the core team. There are some tradeoffs here, not having to opt-in is nice, parsing every JS file is unfortunate (it is opt-in via using this off-by-default renderer though, so that's a bit better).

**Review by matthewp — CHANGES_REQUESTED** [review]
Let's talk about this with the core team before merging.

**Comment by natemoo-re** [comment]
Thanks @matthewp! Totally agree, looking forward to chatting about it on the RFC call!

**Comment by natemoo-re** [comment]
Going to close this and reopen as an RFC, per the RFC call.

**Pull request closed without merging**
